eclipse-collections/src/main/java/org/eclipse/collections/impl/map/immutable/ImmutableMapFactoryImpl.java

```diff
@@ -184,4 +206,58 @@
                 throw new AssertionError();
         }
     }
+
+    /**
+     * @since 10.3.0
+     */
+    @Override
+    public <K, V> ImmutableMap<K, V> ofMapIterable(MapIterable<K, V> mapIterable)
+    {
+        return this.withMapIterable(mapIterable);
+    }
+
+    /**
+     * @since 10.3.0
+     */
+    @Override
+    public <K, V> ImmutableMap<K, V> withMapIterable(MapIterable<K, V> mapIterable)
+    {
+        if (mapIterable.isEmpty())
```

Comment:
The entire implementation is just:
```suggestion
      return (ImmutableMap) this.mapIterable.toImmutable();
```